core/drivers/clk/clk.c

```diff
@@ -119,17 +138,21 @@ TEE_Result clk_register(struct clk *clk)
 
 	clk_compute_rate_no_lock(clk);
 
-	DMSG("Registered clock %s, freq %lu", clk_get_name(clk),
+	DMSG("Registered %sclock %s, freq %lu",
+	     clk_is_clk_orphan(clk) ? "orphan" : "", clk_get_name(clk),
 	     clk_get_rate(clk));
 
 	return TEE_SUCCESS;
 }
 
 const char *clk_get_name(struct clk *clk)
 {
-	if (clk_is_clk_elt(clk))
+	if (clk_is_clk_elt(clk) && clk_to_clk_elt(clk)->name)
 		return clk_to_clk_elt(clk)->name;
 
+	if (clk->ops->get_name)
+		return clk->ops->get_name(clk);
```

Comment:
not addressed yet.

Reply:
addressed in https://github.com/OP-TEE/optee_os/pull/4942/commits/3e89ec12b097109b8a6c207963a66dcd16ea5fc8

Reply:
I'm sorry, this isn't what I was looking for. This function still contains special treatment for struct clk_elt. I'd rather that struct clk_elt always provided a `get_name()`. If that function just returns `clk_to_clk_elt(clk)->name`, that's fine the point is that we shouldn't care here.

Reply:
Would this [fixup commit](https://github.com/OP-TEE/optee_os/pull/4942/commits/532e0867f3fd2095032588b6410d965ccf2c22dd) do the job?
The _ugly_ part, IMO, is that it mandates a specific `get_name` operator for `clk_elt` instances,
instead of _ugly_ routing `clk_get_name()` according to the clock type.

Reply:
It's an improvement. I have a couple of comments though. :-)